test/lit/passes/coalesce-locals-gc-nn.wast

```diff
@@ -0,0 +1,84 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --coalesce-locals -all --enable-gc-nn-locals -S -o - \
+;; RUN:   | filecheck %s
+
+(module
+ ;; CHECK:      (func $nn-locals (param $0 (ref any))
+ ;; CHECK-NEXT:  (local $1 ((ref any) (ref any)))
+ ;; CHECK-NEXT:  (local $2 ((ref any) (ref any)))
+ ;; CHECK-NEXT:  (local.set $1
+ ;; CHECK-NEXT:   (tuple.make
+ ;; CHECK-NEXT:    (local.get $0)
+ ;; CHECK-NEXT:    (local.get $0)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.set $2
+ ;; CHECK-NEXT:   (tuple.make
+ ;; CHECK-NEXT:    (local.get $0)
+ ;; CHECK-NEXT:    (local.get $0)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (call $nn-locals
+ ;; CHECK-NEXT:   (tuple.extract 0
+ ;; CHECK-NEXT:    (local.get $1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (call $nn-locals
+ ;; CHECK-NEXT:   (tuple.extract 0
+ ;; CHECK-NEXT:    (local.get $2)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (call $nn-locals
+ ;; CHECK-NEXT:   (tuple.extract 1
+ ;; CHECK-NEXT:    (local.get $1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (call $nn-locals
+ ;; CHECK-NEXT:   (tuple.extract 1
+ ;; CHECK-NEXT:    (local.get $2)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $nn-locals (param $any (ref any))
+  ;; When computing interferences, coalesce locals should not error on tuples
+  ;; that contain non-nullable locals.
+  (local $x ((ref any) (ref any)))
+  (local $y ((ref any) (ref any)))
```

Comment:
These two locals are set and used in exactly the same way... Aren't they supposed to be "coalesced" by `CoalesceLocals`?

Reply:
Then how about adding a comment in this file that this does not test coalescing tuples atm (despite the file name) but only tests it does not crash?

Reply:
On sorry, line 48 already has it... Nevermind 😅